common/src/main/java/com/wynntils/models/containers/LootChestModel.java

```diff
@@ -192,4 +259,68 @@ private void resetNormalDryStatistics() {
         dryEmeralds.store(0);
         dryItemTiers.store(new EnumMap<>(GearTier.class));
     }
+
+    // region Poi Migration
+    public static boolean isCustomPoiLootChest(CustomPoi customPoi) {
+        boolean nameIsLootChest = Arrays.stream(LootChestTier.values())
+                .anyMatch(tier -> tier.getWaypointName().equals(customPoi.getName()));
+        boolean colorIsLootChest = customPoi.getColor().equals(CommonColors.WHITE);
+        boolean iconIsLootChest = Arrays.stream(LootChestTier.values())
+                .anyMatch(tier -> tier.getWaypointTexture().equals(customPoi.getIcon()));
+        boolean visibilityIsLootChest = customPoi.getVisibility() == CustomPoi.Visibility.DEFAULT;
+
+        return nameIsLootChest && colorIsLootChest && iconIsLootChest && visibilityIsLootChest;
+    }
+
+    private void startPoiMigration() {
+        // The feature instance is not guaranteed to be present, so we have to check
+        MainMapFeature featureInstance = Managers.Feature.getFeatureInstance(MainMapFeature.class);
+        if (featureInstance == null) return;
+
+        Config<List<CustomPoi>> customPois = featureInstance.customPois;
+        if (customPois.get().isEmpty()) return;
+
+        // Try to migrate custom pois to the new mapdata system
+        // This is done on storage load, as configs are loaded before storages
+        List<CustomPoi> migratedPois = new ArrayList<>();
+        for (CustomPoi customPoi : customPois.get()) {
+            if (migrateToMapdata(customPoi)) {
+                migratedPois.add(customPoi);
+            }
+        }
+
+        WynntilsMod.info("MapData Migration: Custom Pois: " + customPois.get().size());
+        WynntilsMod.info("MapData Migration: Found Loot Chests: " + migratedPois.size());
+
+        customPois.get().removeAll(migratedPois);
+        customPois.touched();
+    }
+
+    // This feature ports old custom poi data to the new mapdata system
+    // This is a one-time migration, but can't be removed in the foreseeable future,
+    // so we can keep upfixing old configs
+    private boolean migrateToMapdata(CustomPoi customPoi) {
+        boolean isLootChest = isCustomPoiLootChest(customPoi);
+        if (!isLootChest) return false;
```

Comment:
I'm a bit tired so I might be missing something obvious, but I'll ask anyway. :) You are only migrating the found loot chests? Should not the rest of the custom pois migrate to waypoints?